app/GNSSStatus/Parsing/GNSSData.cs

```diff
@@ -52,50 +102,90 @@ public string GetPayloadJson()
+            TimeUtc = roverUtCTime,
+            //FixType = worstFixType,
+            FixType = roverFixType,
+            //SatellitesInUse = GGA.TotalSatellitesInUse,
+            SatellitesInUse = roverSatInUse,
+            //RoverX = roverXAverage,
+            RoverX = roverXMedian,
+            //RoverY = roverYAverage,
+            RoverY = roverYMedian,
+            //RoverZ = roverZAverage,
+            RoverZ = roverZMedian,
         });
         
         builder.AddPayload(new
         {
-            DeltaXY = deltaXYAverage,
-            DeltaZ = deltaZAverage,
-            PDop = GSA.PDop,
-            HDop = GSA.HDop,
-            VDop = GSA.VDop
+            //DeltaXY = deltaXYAverage,
+            DeltaXY = deltaXy,
+            //DeltaZ = deltaZAverage,
+            DeltaZ = deltaZ,
+            //PDop = GSA.PDop,
+            PDop = pDop,
+            //HDop = GSA.HDop,
+            HDop = hDop,
+            //VDop = GSA.VDop
+            VDop = vDop,
         });
         
         builder.AddPayload(new
         {
             RoverId = ConfigManager.CurrentConfiguration.RoverIdentifier,
-            ErrorLatitude = GST.LatitudeError,
-            ErrorLongitude = GST.LongitudeError,
-            ErrorAltitude = GST.AltitudeError
+            //ErrorLatitude = GST.LatitudeError,
+            ErrorLatitude = errLat,
+            //ErrorLongitude = GST.LongitudeError,
+            ErrorLongitude = errLon,
+            //ErrorAltitude = GST.AltitudeError
+            ErrorAltitude = errAlt
         });
         
         builder.AddPayload(new
         {
             DifferentialDataAge = GGA.AgeOfDifferentialData,
             ReferenceStationId = GGA.DifferentialReferenceStationID,
-            BaseRoverDistance = NTR.DistanceBetweenBaseAndRover,
+            //BaseRoverDistance = NTR.DistanceBetweenBaseAndRover,
+            BaseRoverDistance = baseline,
             IonoPercentage = IonoPercentage
         });
         
         return builder.Build(true);
     }
     
+    public (int, int) CalculateMedianIndex<T>(List<T> list)
+    {
+        // Luo kopio alkuperäisestä listasta ja liitä siihen alkuperäiset indeksit
+        var indexedNumbers = list
+            .Select((value, index) => new { Value = value, Index = index })
+            .OrderBy(x => x.Value)
+            .ToArray();
+
+        int count = indexedNumbers.Length;
+
+        if (count % 2 == 0)
+        {
+            // Jos määrä on parillinen, laske kahden keskimmäisen luvun indeksit
+            int mid1Index = indexedNumbers[count / 2 - 1].Index;
+            int mid2Index = indexedNumbers[count / 2].Index;
+            return (mid1Index, mid2Index);
+        }
+        else
+        {
+            // Jos määrä on pariton, valitse keskimmäisen luvun indeksi
+            int medianIndex = indexedNumbers[count / 2].Index;
+            return (medianIndex, -1); // -1 tarkoittaa, että vain yksi mediaani löytyy
+        }
```

Comment:
Does not work as expected, and returns incorrect indices for generic types that do not implement IComparable.